packages/arb-token-bridge-ui/src/hooks/useAccountType.ts

```diff
@@ -1,46 +1,27 @@
-import { useState, useEffect } from 'react'
 import { useAccount, useProvider } from 'wagmi'
+import useSWRImmutable from 'swr/immutable'
 
 import { addressIsSmartContract } from '../util/AddressUtils'
 
 type Result = {
-  isEOA: boolean | undefined
-  isSmartContractWallet: boolean | undefined
+  isEOA: boolean
+  isSmartContractWallet: boolean
+  isLoading: boolean
 }
 
-const defaultResult: Result = {
-  isEOA: undefined,
-  isSmartContractWallet: undefined
-}
-
-export function useAccountType() {
-  const provider = useProvider()
+export function useAccountType(): Result {
   const { address } = useAccount()
+  const provider = useProvider()
 
-  const [result, setResult] = useState<Result>(defaultResult)
-
-  useEffect(() => {
-    async function update() {
-      if (typeof address === 'undefined') {
-        setResult(defaultResult)
-        return
-      }
-
-      // TODO: Try to detect counterfactual/just-in-time deployed smart contract wallets
-
-      const isSmartContractWallet = await addressIsSmartContract(
-        address,
-        provider
-      )
-
-      setResult({
-        isEOA: !isSmartContractWallet,
-        isSmartContractWallet
-      })
-    }
-
-    update()
-  }, [address, provider])
-
-  return result
+  const { data: isSmartContractWallet = false, isLoading } = useSWRImmutable(
+    address ? [address, provider, 'useAccountType'] : null,
+    ([_address, _provider]) => addressIsSmartContract(_address, _provider)
+  )
+
+  // By default, assume it's an EOA
+  return {
+    isEOA: !isSmartContractWallet,
```

Comment:
This will resolve to `true` while we are loading, whereas if you look at how we used the hook in `AdvancedSettings` we default both to `false`. Maybe we should just default both values to `false` while loading